File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-1.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

 func solve(root *TreeNode , inorder *[]int){
    if root==nil{
        return
    }

    solve(root.Left , inorder)
    *inorder = append(*inorder , root.Val)
    solve(root.Right , inorder)
 }

func kthSmallest(root *TreeNode, k int) int {
    var inorder []int
    solve(root , &inorder)

    return inorder[k-1]
}

File: Data Structures & Algorithms/balanced-binary-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int maxHeight(TreeNode* root)
    {
        if(root==NULL)
            return 0;

        int left = maxHeight(root->left);
        int right = maxHeight(root->right);

        return 1 + max(left , right);
    }
    bool isBalanced(TreeNode* root) {
        if (root ==NULL)
            return  true;
       bool left =  isBalanced(root->left);
       bool right = isBalanced(root->right);
        int leftH = maxHeight(root->left);
        int rightH = maxHeight(root->right);

        if (abs(leftH - rightH)>1)
            return false;

        return left && right ;
        
    }
};
